src/auth/strategies/discord.strategy.ts

```diff
@@ -0,0 +1,42 @@
+import { Inject, Injectable } from "@nestjs/common";
+import { PassportStrategy } from "@nestjs/passport";
+import { Strategy } from "passport-discord";
+import { Profile } from "passport-discord";
+import { IAuthProvider } from "../../global/interfaces/oauth.interface";
+
+@Injectable()
+export class DiscordStrategy extends PassportStrategy(Strategy, "discord") {
+    constructor(
+        @Inject("DISCORD_OAUTH")
+        private readonly discordAuthService: IAuthProvider,
+    ) {
+        super({
+            clientID: process.env.DISCORD_CLIENT_ID,
+            clientSecret: process.env.DISCORD_CLIENT_SECRET,
+            callbackURL: process.env.DISCORD_CALLBACK_URL,
+            scope: ["identify"],
+        });
+    }
```

Comment:
I'm getting an error saying `context.$queryRaw is not a function`
![Screenshot 2024-08-11 at 6 48 27 PM](https://github.com/user-attachments/assets/a60b414a-0723-4b5e-8bc5-27ee621223d1)